File: backend/src/order/objects/OrderChatMessage.scala
package delivery.order.objects

import delivery.shared.objects.OrderId

final case class OrderChatMessage(
    id: String,
    orderId: OrderId,
    senderRole: String,
    peerRole: String,
    messageType: String,
    content: String,
    createdAt: String,
    readAt: Option[String] = None
)

File: frontend/src/objects/order/OrderChatMessage.ts
import type { OrderId } from '@/objects/shared/ids'

export type OrderChatRole = 'customer' | 'merchant' | 'rider'
export type OrderChatMessageType = 'text' | 'image'

export interface OrderChatMessage {
  id: string
  orderId: OrderId
  senderRole: OrderChatRole
  peerRole: OrderChatRole
  messageType: OrderChatMessageType
  content: string
  createdAt: string
}
